accounts-db/src/accounts_db.rs

```diff
@@ -3104,10 +3096,19 @@ impl AccountsDb {
             .sum::<usize>() as u64
     }
 
-    /// Construct a vec of pubkeys for cleaning from:
-    ///   uncleaned_pubkeys - the delta set of updated pubkeys in rooted slots from the last clean
-    ///   dirty_stores - set of stores which had accounts removed or recently rooted
-    /// returns the minimum slot we encountered
+    fn insert_pubkey(&self, candidates: &[RwLock<HashMap<Pubkey, CleaningInfo>>], pubkey: Pubkey) {
+        let index = self.accounts_index.bin_calculator.bin_from_pubkey(&pubkey);
+        let mut candidates_bin = candidates[index].write().unwrap();
```

Comment:
nit: maybe we could sort and group the pubkeys by bins and insert all pubkeys to the same bin at once. 
This will save the repeated costs of lock and unlock of the same bin, also we may have better cache locality.

Reply:
I’ll make a follow-up PR. Thank you.